tools/ci_build/github/linux/docker/scripts/install_deps.sh

```diff
@@ -1,23 +1,53 @@
 #!/bin/bash
 set -e
 
-SYS_LONG_BIT=$(getconf LONG_BIT)
+while getopts p:d: parameter_Option
+do case "${parameter_Option}"
+in
+p) PYTHON_VER=${OPTARG};;
+d) DEVICE_TYPE=${OPTARG};;
+esac
+done
+
+echo "Python version=$PYTHON_VER"
+
+DEVICE_TYPE=${DEVICE_TYPE:=Normal}
 
-echo "Installing azcopy"
+
+SYS_LONG_BIT=$(getconf LONG_BIT)
+mkdir -p /tmp/src
 if [ $SYS_LONG_BIT = "64" ]; then
+  echo "Installing azcopy"
   mkdir -p /tmp/azcopy
   aria2c -q -d /tmp/azcopy -o azcopy.tar.gz https://aka.ms/downloadazcopy-v10-linux
   tar --strip 1 -xf /tmp/azcopy/azcopy.tar.gz -C /tmp/azcopy
   cp /tmp/azcopy/azcopy /usr/bin
+  echo "Installing cmake"
+  aria2c -q -d /tmp/src "https://github.com/Kitware/CMake/releases/download/v3.15.4/cmake-3.15.4-Linux-x86_64.tar.gz"
+  tar -zxf /tmp/src/cmake-3.15.4-Linux-x86_64.tar.gz --strip=1 -C /usr
+else
+  echo "Installing cmake"
+  aria2c -q -d /tmp/src https://github.com/Kitware/CMake/releases/download/v3.15.4/cmake-3.15.4.tar.gz
+  tar -xf /tmp/src/cmake-3.15.4.tar.gz -C /tmp/src
+  cd /tmp/src/cmake-3.15.4
+  ./configure --prefix=/usr --parallel=`nproc` --system-curl --system-zlib --system-expat
```

Comment:
>./configure --prefix=/usr --parallel=`nproc` [](start = 2, length = 44)

I was wondering why do we need this?

Reply:
For x86_32, cmake doesn't provide a prebuilt binary.

Reply:
They are probably right.

---
In reply to: [331165249](https://github.com/microsoft/onnxruntime/pull/1986#discussion_r331165249) [](ancestors = 331165249)